src/main/java/com/newrelic/telemetry/opentelemetry/export/NewRelicMetricExporter.java

```diff
@@ -0,0 +1,186 @@
+package com.newrelic.telemetry.opentelemetry.export;
+
+import com.newrelic.telemetry.Attributes;
+import com.newrelic.telemetry.SimpleMetricBatchSender;
+import com.newrelic.telemetry.TelemetryClient;
+import com.newrelic.telemetry.metrics.Metric;
+import com.newrelic.telemetry.metrics.MetricBatchSenderBuilder;
+import com.newrelic.telemetry.metrics.MetricBuffer;
+import io.opentelemetry.sdk.internal.MillisClock;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor.Type;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.util.Collection;
+
+public class NewRelicMetricExporter implements MetricExporter {
+
+  private final Attributes commonAttributes;
+  private final TelemetryClient telemetryClient;
+  private final TimeTracker timeTracker;
+  private final MetricPointAdapter metricPointAdapter;
+
+  public NewRelicMetricExporter(
+      TelemetryClient telemetryClient,
+      Attributes serviceAttributes,
+      TimeTracker timeTracker,
+      MetricPointAdapter metricPointAdapter) {
+    this.telemetryClient = telemetryClient;
+    this.timeTracker = timeTracker;
+    // todo: these two attributes are the same as the ones in the SpanBatchAdapter. Move to
+    // somewhere common.
+    this.commonAttributes =
+        serviceAttributes
+            .copy()
+            .put("instrumentation.provider", "opentelemetry")
+            .put("collector.name", "newrelic-opentelemetry-exporter");
+    this.metricPointAdapter = metricPointAdapter;
+  }
+
+  public static Builder newBuilder() {
+    return new Builder();
+  }
+
+  @Override
+  public ResultCode export(Collection<MetricData> metrics) {
+    MetricBuffer buffer = MetricBuffer.builder().attributes(commonAttributes).build();
+    for (MetricData metric : metrics) {
+      Descriptor descriptor = metric.getDescriptor();
+      Type type = descriptor.getType();
+
+      Attributes attributes = buildCommonAttributes(metric);
+
+      Collection<Point> points = metric.getPoints();
+      for (Point point : points) {
+        Collection<Metric> metricsFromPoint =
+            metricPointAdapter.buildMetricsFromPoint(descriptor, type, attributes.copy(), point);
+        metricsFromPoint.forEach(buffer::addMetric);
+      }
+    }
+    timeTracker.tick();
+    telemetryClient.sendBatch(buffer.createBatch());
+    return ResultCode.SUCCESS;
+  }
+
+  @Override
+  public void shutdown() {
+    telemetryClient.shutdown();
+  }
+
+  private Attributes buildCommonAttributes(MetricData metric) {
+    Attributes attributes = new Attributes();
+    CommonUtils.addResourceAttributes(attributes, metric.getResource());
+    CommonUtils.populateLibraryInfo(attributes, metric.getInstrumentationLibraryInfo());
```

Comment:
these two methods from `CommonUtils` are called back-to-back in both places in production code.  This might indicate that they should be one, uh, common thing.  Since there's not readily an interface for "has-resource-and-library-info" that is common to spandata and metricdata, you might consider a method that takes 3 args to combine.  I dunno.  You might be able to also encapsulate the `new Attributes()` creation too.

Reply:
That method name would be pretty fugly. I agree it's a little redundant, but it also makes it very clear what each method is doing, in case we find a bug in one or the other, it's a little easier to track down, maybe?

Reply:
Yeah, I don't have a great name to offer up either.  😄 Let's just go with it for now.